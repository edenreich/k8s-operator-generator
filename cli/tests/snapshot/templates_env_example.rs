use askama::Template;
use insta::assert_snapshot;
use kopgen::{errors::AppError, templates::general::EnvExample};

#[test]
fn render_env_example() -> Result<(), AppError> {
    let template = EnvExample {
        operator_name: "example_operator".to_string(),
        operator_author: "Author Name".to_string(),
        operator_api_group: "example.com".to_string(),
        operator_api_version: "v1".to_string(),
        operator_resource_ref: "resourceRef".to_string(),
        operator_example_metadata_spec_field_ref: "fieldRef".to_string(),
        operator_include_tags: "tag1".to_string(),
        operator_secret_name: "secret".to_string(),
    };
    let rendered = template.render()?;
    assert_snapshot!(rendered);
    Ok(())
}

#[test]
fn render_env_example_with_empty_string() -> Result<(), AppError> {
    let template = EnvExample {
        operator_name: "".to_string(),
        operator_author: "".to_string(),
        operator_api_group: "".to_string(),
        operator_api_version: "".to_string(),
        operator_resource_ref: "".to_string(),
        operator_example_metadata_spec_field_ref: "".to_string(),
        operator_include_tags: "".to_string(),
        operator_secret_name: "".to_string(),
    };
    let rendered = template.render()?;
    assert_snapshot!(rendered);
    Ok(())
}
