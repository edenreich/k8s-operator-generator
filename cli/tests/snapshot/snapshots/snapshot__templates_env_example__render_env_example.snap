---
source: cli/tests/snapshot/templates_env_example.rs
expression: rendered
---
KUBERNETES_OPERATOR_NAME=example_operator
KUBERNETES_OPERATOR_AUTHOR=Author Name
KUBERNETES_OPERATOR_API_GROUP=example.com
KUBERNETES_OPERATOR_API_VERSION=v1
KUBERNETES_OPERATOR_RESOURCE_REF=resourceRef
KUBERNETES_OPERATOR_EXAMPLE_METADATA_SPEC_FIELD_REF=fieldRef
KUBERNETES_OPERATOR_INCLUDE_TAGS=tag1
KUBERNETES_OPERATOR_SECRET_NAME=secret

OPENAPI_DOWNLOAD_URL=https://raw.githubusercontent.com/edenreich/kopgen/refs/heads/main/openapi.yaml
INSTALL_CRDS=true
RUST_LOG=info
RUST_BACKTRACE=1
CONTAINER_REGISTRY=localhost:5005
CLUSTER_NAME=k3d-k3s-default

#TARGET_ARCH=x86_64-unknown-linux-musl
RELEASE=false
